controllers/nginx/pkg/cmd/controller/nginx.go

```diff
@@ -92,15 +93,48 @@ type NGINXController struct {
 // Start start a new NGINX master process running in foreground.
 func (n NGINXController) Start() {
 	glog.Info("starting NGINX process...")
+
+	done := make(chan error, 1)
 	cmd := exec.Command(n.binary, "-c", cfgPath)
+	n.start(cmd, done)
+	select {
+	case err := <-done:
+		if exitError, ok := err.(*exec.ExitError); ok {
```

Comment:
I will add the comment in the code. Basically if the nginx master process dies the workers continue to process requests (passing the checks) but in case of updates in ingress no updates will be reflected in the nginx configuration